frontend/src/components/profile/profileSettings/avatarSettings/handleAvatarUpload.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Converts the passed file to a Base64 string while applying some
+ * validations to it.
+ * @param file
+ * @returns a Promise that resolves to a Base64 string. If the uploaded
+ * file is not an image, is over 2MB, or the upload fails for any reason,
+ * this function will reject with a corresponding error message (a string).
+ */
+export const handleAvatarUpload = async (file: File) =>
```

Comment:
The upload works very well, good job! However, I noticed one issue. When you have already uploaded a file, and you go to browse, if you just select a file but close the modal, instead of uploading - it crashes with errors in the console:
![image](https://github.com/julkascript/cardflow/assets/59143945/2936029c-6d13-44bf-9930-8a4515c7b4af)

However, you do not have to fix this issue now. I will add it as an `issue` to the issues list in the repo.